src/hotspot/share/oops/instanceKlass.cpp

```diff
@@ -770,20 +770,43 @@ void InstanceKlass::link_class(TRAPS) {
 void InstanceKlass::check_link_state_and_wait(JavaThread* current) {
   MonitorLocker ml(current, _init_monitor);
 
+  bool debug_logging_enabled = log_is_enabled(Debug, class, init);
+
   // Another thread is linking this class, wait.
   while (is_being_linked() && !is_init_thread(current)) {
+    if (debug_logging_enabled) {
+      ResourceMark rm(current);
+      log_debug(class, init)("Thread %s waiting for linking of %s by thread %s",
```

Comment:
Here and later: Should thread names be in quotes? This would match other places, e.g. error handler, jstack output, etc.

Reply:
I don't think it is critical but I can change it.